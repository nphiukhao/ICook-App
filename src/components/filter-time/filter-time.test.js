import React, {Component} from 'react'
import FilterTime from './filter-time'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PropTypes from 'prop-types'


class ContextProvider extends Component {

    static childContextTypes = {
        myContext: PropTypes.object
    }

    getChildContext = () => ({
        myContext: {
            recipeArray: [{
                date_modified: "2019-10-17T10:45:28.356Z",
                id: 1,
                instructions: "Cook fettuccini in a pot",
                minutes: 35,
                title: "Chicken Fettuccine Alfredo",
            },{
                date_modified: "2019-10-17T10:45:28.356Z",
                id: 2,
                instructions: "Cook fettuccini in a pot",
                minutes: 35,
                title: "Chicken Fettuccine Alfredo",
            },
            ]
        }
    })
    render(){
        return this.props.children
      }
}


describe(`Filter by Time component`, () => {
    it('filter by time snapshot test', () => {
        const tree = shallow(<ContextProvider><FilterTime/></ContextProvider>)
        expect(toJson(tree)).toMatchSnapshot()
    })
})