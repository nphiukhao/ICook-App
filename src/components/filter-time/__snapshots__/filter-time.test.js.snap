// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter by Time component filter by time snapshot test 1`] = `<FilterTime />`;
